stan/math/prim/fun/symmetrize.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef STAN_MATH_PRIM_FUN_SYMMETRIZE_HPP
+#define STAN_MATH_PRIM_FUN_SYMMETRIZE_HPP
+
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return a symmetric matrix using elements from the lower triangular part of
+ * the input matrix.
+ *
+ * @tparam T type of elements in the matrix
+ * @param m Matrix.
+ * @throw std:invalid_argument if the matrix is not square.
+ */
+template <typename T, require_eigen_t<T>* = nullptr>
+inline Eigen::Matrix<value_type_t<T>, Eigen::Dynamic, Eigen::Dynamic>
+symmetrize(const T& m) {
```

Comment:
Should we maybe rename this to `symmetrize_low` or `symmetrize_to_upp` (not sure which one would be more descriptive) and then implement the upper versions as well (though that can be a separate PR).

Reply:
I think you're right that it should be named differently, I just don't know a good non-verbose name for it. Maybe, `symmetrize_lower_tri`?

Reply:
That seems like a good name to me. So `symmetrize_lower_tri()` would mean the lower triangular matrix gets copied to upper triangular.

Reply:
@t4c1 suggested `symmetrize_from_lower_tri()`. What do you think?

Reply:
Either one is good for me. I like the first one best for clarity.

Reply:
Ok, lets go with that then.